modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectViewFilterColumnLocalServiceImpl.java

```diff
@@ -126,16 +129,31 @@ private void _validate(
 						"\" is not filterable"));
 			}
 
-			if (Validator.isNull(objectViewFilterColumn.getFilterType()) &&
-				Validator.isNull(objectViewFilterColumn.getJSON())) {
+			String filterType = objectViewFilterColumn.getFilterType();
 
+			boolean filterTypeNull = Validator.isNull(filterType);
+
+			String json = objectViewFilterColumn.getJSON();
+
+			boolean jsonNull = Validator.isNull(json);
+
+			if (filterTypeNull && jsonNull) {
 				continue;
 			}
 
-			if ((Validator.isNull(objectViewFilterColumn.getFilterType()) &&
-				 Validator.isNotNull(objectViewFilterColumn.getJSON())) ||
-				(Validator.isNotNull(objectViewFilterColumn.getFilterType()) &&
-				 Validator.isNull(objectViewFilterColumn.getJSON()))) {
+			boolean jsonValid = JSONUtil.isJSONObject(json);
+
+			if (jsonValid) {
+				JSONObject jsonObject = _jsonFactory.createJSONObject(json);
+
+				String value = String.valueOf(jsonObject.get(filterType));
+
+				jsonValid = Validator.isNotNull(
+					value.replaceAll("[\\[\\]\"]", ""));
+			}
+
+			if ((filterTypeNull && !jsonNull) ||
+				(!filterTypeNull && jsonNull) || !jsonValid) {
```

Comment:
@rijema no need to create new variables here, please simplify by checking if the JSONArray is empty:

```
if (Validator.isNull(objectViewFilterColumn.getFilterType()) &&
	Validator.isNull(objectViewFilterColumn.getJSON())) {

	continue;
}

if ((Validator.isNull(objectViewFilterColumn.getFilterType()) &&
	 Validator.isNotNull(objectViewFilterColumn.getJSON())) ||
	(Validator.isNotNull(objectViewFilterColumn.getFilterType()) &&
	 Validator.isNull(objectViewFilterColumn.getJSON()))) {

	throw new ObjectViewFilterColumnException(
		StringBundler.concat(
			"Object field name \"",
			objectViewFilterColumn.getObjectFieldName(),
			"\" needs to have the filter type and JSON specified"));
}

JSONObject jsonObject = _jsonFactory.createJSONObject(
	objectViewFilterColumn.getJSON());

if (JSONUtil.isEmpty(
		jsonObject.getJSONArray(
			objectViewFilterColumn.getFilterType()))) {

	throw new ObjectViewFilterColumnException(
		StringBundler.concat(
			"Object field name \"",
			objectViewFilterColumn.getObjectFieldName(),
			"\" needs to have the filter type and JSON specified"));
}
```

Reply:
Hi. This doesn't cover the scenario we're the customer is passing a value like {"includes": [""]}, only {"includes": []}. How should we proceed?

Reply:
@rijema you should create a test with this case and adapt the code